aop/src/main/java/io/micronaut/aop/writer/AopProxyWriter.java

```diff
@@ -527,6 +528,31 @@ public void visitIntroductionMethod(TypedElement declaringBean,
     public void visitAroundMethod(TypedElement beanType,
                                   MethodElement methodElement) {
 
+        final List<MethodElement> overridden = methodElement.getOwningType()
+                .getEnclosedElements(ElementQuery.ALL_METHODS
+                        .filter(el -> el.getName().equals(methodElement.getName()) && el.overrides(methodElement)));
```

Comment:
add `.named(n -> n.equals(methodElement.getName())` instead of the 2 conditions in the filter since it will be more efficient as it won't need to materialize the `MethodElement`